users/ignoring.rst

```diff
@@ -88,6 +88,32 @@ The ``.stignore`` file contains a list of file or path patterns. The
    are *included* (that is, *not* ignored). This can be used to override
    more general patterns that follow.
 
+   .. note::
+
+      Negated patterns that can match items below the folder root will cause
+      Syncthing to traverse otherwise ignored directories. If the
+      :ref:`watcher <scanning>` is enabled, those directories will also be
+      watched. Directories ignored before the first negated pattern can
+      however be safely skipped, since the first matching pattern wins. For
+      example::
+
+         /foo
+         /bar
+         !baz
+         *
+
+      The directories ``foo`` and ``bar`` will be entirely ignored. However any
+      other directories present must be scanned entirely to find any items
+      named `baz`, despite the fact that they will be ignored due to the
+      ``*``. As a special case, top-level rooted patterns (e.g. ``!/foo``) do
+      not cause this behaviour::
```

Comment:
So far "this behavior" (U.S. English seems to be used mostly in the docs) is not very well understood, which makes it hard to see what exactly top-level rooted patterns really do.  They're obviously negated, but don't cause something which essentially meant not doing something.  So do they do it or don't they? :exploding_head:

And "top-level rooted patterns" is a bit too terse for parsing correctly IMO, even with the example.  The example is also negated btw. so does the remark apply to all top-level rooted negated patterns or what?